internal/controller/pod_controller_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"testing"
+
+	cloudsqlapi "github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/api/v1alpha1"
+	"github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/testhelpers"
+	"github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/workload"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func TestPodWebhookWithDeploymentOwners(t *testing.T) {
+	_, scheme, err := clientBuilder()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Proxy workload
+	p := testhelpers.BuildAuthProxyWorkload(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "project:region:db")
+	addFinalizers(p)
+	addSelectorWorkload(p, "Deployment", "app", "webapp")
+
+	// Deployment that matches the proxy
+	dMatch := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dMatch.ObjectMeta.Labels = map[string]string{"app": "webapp"}
+
+	// Deployment that does not match the proxy
+	dNoMatch := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dNoMatch.ObjectMeta.Labels = map[string]string{"app": "other"}
+
+	// Deployment that matches the proxy with an owner of an unknown kind
+	dWithOwner := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dWithOwner.ObjectMeta.Labels = map[string]string{"app": "webapp"}
+	deploymentOwner := &v1.PartialObjectMetadata{
+		TypeMeta:   v1.TypeMeta{Kind: "BuildToolOperatror", APIVersion: "v1"},
```

Comment:
sp: BuildToolOperatror

nit: Maybe just "DontTouchThisThing" or something obviously not of a known kind. BuildToolOperator looks legit.

Reply:
Fixed. I added more comments to make this easier to understand.